### PR TITLE
#4505: Add end to end demo for functional bert model

### PR DESCRIPTION
**Pull request by Sudharsan-V**
This PR has end to end demo for the functional Bert model.

### CODE REVIEW

File: models/experimental/functional_bert/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## functional_bert Demo
+## How to Run
+
+Use `pytest --disable-warnings models/experimental/functional_bert/demo/demo.py::test_demo[models.experimental.functional_bert.tt.ttnn_functional_bert-384-8-phiyodr/bert-large-finetuned-squad2-models/experimental/functional_bert/demo/input_data.json]` to run the demo.
```
Comment (boris-drazic) [inline models/experimental/functional_bert/README.md:4]: Can we change how the test is invoked to only expose parameters that users are allowed to change.
Here we want the demo to run with 384 seq size and 8 batch size in all cases for bert large, so this should not be part of test parameters.
The only parameter here should be path to input data.

File: models/experimental/functional_bert/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+If you wish to run the demo with a different input use `pytest --disable-warnings models/experimental/functional_bert/demo/demo.py::test_demo[models.experimental.functional_bert.tt.ttnn_functional_bert-384-8-phiyodr/bert-large-finetuned-squad2-<address_to_your_json_file.json-1>]`. This file is expected to have exactly 8 inputs.
```
Comment (boris-drazic) [inline models/experimental/functional_bert/README.md:6]: `<address_to_your_json_file.json-1>` should be without `-1` at the end.

Can you check this. I get the following error using a different input file
```
ERROR: not found: /home/boris/tt-metal/models/experimental/functional_bert/demo/demo.py::test_demo[models.experimental.functional_bert.tt.ttnn_functional_bert-384-8-phiyodr/bert-large-finetuned-squad2-models/experimental/functional_bert/demo/input_data22.json
(no name '/home/boris/tt-metal/models/experimental/functional_bert/demo/demo.py::test_demo[models.experimental.functional_bert.tt.ttnn_functional_bert-384-8-phiyodr/bert-large-finetuned-squad2-models/experimental/functional_bert/demo/input_data22.json' in any of [<Module models/experimental/functional_bert/demo/demo.py>])
```

**Review by boris-drazic — CHANGES_REQUESTED** [review]

**Review by boris-drazic — APPROVED** [review]

File: models/experimental/functional_bert/demo/demo.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    # set up tokenizer
+    tokenizer_name = str(model_location_generator(model_name, model_subdir="Bert"))
+    tokenizer = BertTokenizer.from_pretrained(tokenizer_name)
+    config = hugging_face_reference_model.config
+    nlp = pipeline("question-answering", model=hugging_face_reference_model, tokenizer=tokenizer)
+
+    if functional_bert == ttnn_functional_bert:
+        tt_model_name = f"ttnn_{model_name}"
+    elif functional_bert == ttnn_optimized_functional_bert:
+        tt_model_name = f"ttnn_optimized_{model_name}"
```
Comment (arakhmati) [inline models/experimental/functional_bert/demo/demo.py:71]: Please change this to `ttnn_{model_name}_optimized`

**Review by arakhmati — CHANGES_REQUESTED** [review]

File: models/experimental/functional_bert/demo/demo.py (hunk 3 above)
Comment (Sudharsan-V) [reply models/experimental/functional_bert/demo/demo.py]: `f"ttnn_optimized_{model_name}"` is changed to `ttnn_{model_name}_optimized`

**Review by arakhmati — APPROVED** [review]

**Comment by Sudharsan-V** [comment]
The CI tests are passing. Hence merged with the main.

**Pull request merged**